src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/aspnetcore-https.js

```diff
@@ -0,0 +1,40 @@
+// This script sets up HTTPS for the application using the ASP.NET Core HTTPS certificate
+const fs = require('fs');
+const spawn = require('child_process').spawn;
+const path = require('path');
+
+const baseFolder =
+  process.env.APPDATA !== undefined && process.env.APPDATA !== ''
+    ? `${process.env.APPDATA}/ASP.NET/https`
+    : `${process.env.HOME}/.aspnet/https`;
+
+const certificateName = process.argv.map(arg => arg.match(/--name=(?<value>.+)/i))
+                                    .filter(Boolean)
+                                    .reduce((previous, current) => previous || current.groups.value, undefined) ||
+  process.env.npm_package_name;
```

Comment:
Some slightly funky formatting going on here. Also if you're interested, this can be done without `reduce` which would reduce the scariness factor a bit:

```suggestion
const certificateArg = process.argv.map(arg => arg.match(/--name=(?<value>.+)/i)).filter(Boolean)[0];
const certificateName = certificateArg ? certificateArg.groups.value : process.env.npm_package_name;
```

The ternary could be reduced further to `certificateArg?.groups.value || process.env.npm_package_name` but I think that's harder to understand.

Since this is just code style please feel free to ignore the suggestion!

Reply:
`?.` and stuff like that is not a good idea in these script files since they need to run in node, but I get your point.

I didn't know you could index on an empty array in JS, that's tricky but neat!